internal/state/dataplane/configuration_test.go

```diff
@@ -136,12 +139,32 @@ func TestBuildConfiguration(t *testing.T) {
 		return r
 	}
 
+	createExpBackendGroupsForRoute := func(route *graph.Route) []BackendGroup {
+		groups := make([]BackendGroup, 0)
+
+		for idx, r := range route.Rules {
+			var backends []Backend
+			if r.ValidFilters && r.ValidMatches {
+				backends = []Backend{expValidBackend}
+			}
+
+			groups = append(groups, BackendGroup{
+				Backends: backends,
+				Source:   client.ObjectKeyFromObject(route.Source),
+				RuleIdx:  idx,
+			})
+		}
+
+		return groups
+	}
+
 	createTestResources := func(name, hostname, listenerName string, paths ...string) (
-		*v1beta1.HTTPRoute, []graph.BackendGroup, *graph.Route,
+		*v1beta1.HTTPRoute, []BackendGroup, *graph.Route,
 	) {
 		hr := createRoute(name, hostname, listenerName, paths...)
 		route := createInternalRoute(hr, paths)
-		return hr, route.GetAllBackendGroups(), route
+		groups := createExpBackendGroupsForRoute(route)
+		return hr, groups, route
 	}
 
 	hr1, hr1Groups, routeHR1 := createTestResources("hr-1", "foo.example.com", "listener-80-1", "/")
```

Comment:
since `hr1Groups` become expected result now (and similar variables), would it make sense to rename it like `expHR1Groups`?